compiler/ballerina-lang/src/main/java/org/ballerinalang/util/diagnostic/Diagnostic.java

```diff
@@ -17,6 +17,9 @@
 */
 package org.ballerinalang.util.diagnostic;
 
+import io.ballerina.tools.diagnostics.Location;
+import org.ballerinalang.model.elements.PackageID;
+
```

Comment:
Shall we remove this interface? The changes we are introducing with this PR are breaking-changes anyway, so I feel its good to do all these changes at one go.

Lets remove `DiagnosticListener.class` also

Reply:
Do you mean the `DiagnosticPosition` interface or `Diagnostic` interface?.

Reply:
both

Reply:
Removed the whole `Diagnostic` interface. Shall I changing all `DiagnosticKind` usages to `DiagnosticSeverity`?

Reply:
+1